### PR TITLE
feat: Implement Comprehensive Integration Tests

### PR DESCRIPTION
**Pull request by kwsantiago**
# Implement Comprehensive Integration Tests

Closes https://github.com/Routstr/proxy/issues/62

## Overview

Implements comprehensive integration tests that startup the proxy server and test all API endpoints with a testmint Cashu wallet, including database state verification.

## What's Added

**Test Infrastructure**
- TestmintWallet for mock Cashu operations 
- DatabaseSnapshot for state change validation
- Test utilities for performance, concurrency, and response validation

**Test Coverage**
- All API endpoints from `/docs` 
- Wallet operations (topup, refund, authentication)
- Proxy forwarding (GET/POST endpoints)
- Provider management and discovery
- Background tasks and error handling
- Performance and load testing

**Key Features**
- Database state verification for all operations
- Concurrent operation testing
- Edge case and error scenario coverage
- Performance requirements validation (< 500ms p95)
- Support for both mock and real testmint instances

## Files Added

```
tests/integration/
├── conftest.py                    # Test fixtures and infrastructure
├── utils.py                       # Test utilities and validators
├── test_*.py                      # 13 test modules covering all endpoints
├── setup_cashu_mint.sh           # Real testmint setup script
└── .env.example                   # Integration test configuration
```

## Running Tests

```bash
# All integration tests
pytest tests/integration/ -v

# Skip slow tests  
pytest tests/integration/ -m "not slow" -v
```

## Test Results

- 159+ integration tests covering all functionality
- 100% API endpoint coverage
- Database consistency validation
- Performance benchmarks established

### CODE REVIEW

File: router/payment/models.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from pathlib import Path
 
 from fastapi import APIRouter
-from pydantic.v1 import BaseModel
+from pydantic import BaseModel
```
[the remark this answers is not in the document]
Comment (kwsantiago) [reply router/payment/models.py]: Ah gotcha, I can take care of that.
